[2.5.4] ClassCastException for recently added DefaultHttpFilters

### Play Version (2.5.x / etc)

2.5.4

### API (Scala / Java / Neither / Both)

Java

### Operating System (Ubuntu 15.10 / MacOS 10.10 / Windows 10)

MacOS 10.10.5

### JDK (Oracle 1.8.0_72, OpenJDK 1.8.x, Azul Zing)

Oracle 1.8.0_92-b14

### Expected Behavior

1. Created a Filter object extending DefaultHttpFilters
2. Injected CORSFilter
3. Called super with CORSFilter 
4. Expecting CORSFilter to "activate"

### Actual Behavior

1. Instead of CORSFilter activating, I receive an exception. See below.

Caused by: java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [Lplay.mvc.EssentialFilter;
	at play.http.DefaultHttpFilters.<init>(DefaultHttpFilters.java:15)
	at global.Filters.<init>(Filters.java:10)

### Reproducible Test Case

To reproduce, simply follow the CORSFilter instructions listed: https://www.playframework.com/documentation/2.5.x/CorsFilter

### Solution

I was able to modify DefaultHttpFilters.java to resolve the casting exception. I'll put together a pull request and reference this issue.

--- a/framework/src/play/src/main/java/play/http/DefaultHttpFilters.java
+++ b/framework/src/play/src/main/java/play/http/DefaultHttpFilters.java
@@ -12,7 +12,7 @@ public class DefaultHttpFilters implements HttpFilters {
   private final EssentialFilter[] filters;
 
   public DefaultHttpFilters(play.api.mvc.EssentialFilter... filters) {
-    this.filters = (EssentialFilter[]) Arrays.stream(filters).map(f -> f.asJava()).toArray();
+    this.filters = Arrays.stream(filters).map(f -> f.asJava()).toArray(EssentialFilter[]::new);
   }
 
   @Override

